test-suite/Math/NumberTheory/TestUtils/Compose.hs

```diff
@@ -0,0 +1,63 @@
+-- |
+-- Module:      Math.NumberTheory.TestUtils.Compose
+-- Copyright:   (c) 2016 Andrew Lelechenko
+-- Licence:     MIT
+-- Maintainer:  Andrew Lelechenko <andrew.lelechenko@gmail.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- Utils to test Math.NumberTheory
+--
+
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans               #-}
+#if __GLASGOW_HASKELL__ >= 800
+{-# OPTIONS_GHC -fno-warn-redundant-constraints #-}
+#endif
+
+module Math.NumberTheory.TestUtils.Compose where
+
+#if MIN_VERSION_base(4,8,0)
+#else
+import Control.Applicative
+#endif
+import Control.Arrow
+import Data.Functor.Classes
+import Data.Functor.Compose
+
+import Test.Tasty.QuickCheck (Arbitrary(..))
+import Test.SmallCheck.Series (Serial(..))
+
+instance Num (f (g a)) => Num (Compose f g a) where
```

Comment:
The moment we have orphans for stuff like this I feel like we should just copy compose into our side and have it be a fresh definition.  Else we will get maintenance sadness at some point.